### ISSUE TITLE
IllegalArgumentException when loading prefab (presumably with bad data)

### ISSUE THREAD

**Issue by fnuecke** [MovingBlocks/Terasology#2110]
Crash log: http://pastebin.com/L9gQtaRn

I can reproduce this 100% with the list of modules seen in the log; If I read the log correctly it looks like it might be a misconfiguration in Rails? Or more generally, some prefab with a rigid body component and some invalid data for the `mass` field. Even so, I'm assuming this should not crash the engine ;)

I haven't gotten around to set up a dev env, so I couldn't verify, but browsing the related code locations, it seems the `null` returned [from the FloatTypeHandler](https://github.com/MovingBlocks/Terasology/blob/develop/engine/src/main/java/org/terasology/persistence/typeHandling/coreTypes/FloatTypeHandler.java#L46) for invalid data is applied to a primitive field [in the Serializer](https://github.com/MovingBlocks/Terasology/blob/develop/engine/src/main/java/org/terasology/persistence/typeHandling/Serializer.java#L97)? I'm sure there's a reason for this (which I'd be interested to hear), but wouldn't returning zero (`0`) as a fallback/default value for number types make more sense than `null` anyway?

**Comment by Cervator** [comment]
Yep this one is Rails for sure, which is presently broken and excluded from the stable module lineup :-)

SkySom (on our IRC) has been working on fixing it up but is stuck on one of those physics issues. It hasn't worked for quite some while now as the original author disappeared.

In any case good point on this crashing the engine when we probably should be catching it somehow. Sounds reasonable to have the FloatTypeHandler return 0 instead of null. For the original reasoning I'd have to defer to @immortius or maybe see if @flo has an idea.

**Comment by soloturn** [comment]
such a thing not happens any more.
